### PR TITLE
Provide an option for camera mirror.

### PR DESCRIPTION
**Pull request by siong1987**
This PR will provide a new option for `CameraMirror`.

- Auto: Only set mirror on when using the front camera, this will make sure that the resulting media(video/picture) is flipped properly.
- On: Always mirror.
- Off: No mirror.

### CODE REVIEW

**Comment by zvonicek** [comment]
Thank you, I just need this. I recommend to merge this PR.

**Comment by siong1987** [comment]
@omergul123 what do you think about this?  i can fix the conflict for this if this is sensible for you.

**Comment by omergul123** [comment]
@slong1987 looks quite good. Can you rebase it (probably the conflicts will disappear) and also add LL prefix to the mirror enums just like LLCameraFlash and LLCameraPosition

**Comment by siong1987** [comment]
done!

File: LLSimpleCamera/LLSimpleCamera.m [hunk 1]
```diff
@@ -527,6 +578,20 @@ - (void)setCameraPosition:(LLCameraPosition)cameraPosition
     
     self.videoCaptureDevice = device;
     self.videoDeviceInput = videoInput;
+
+    if (_position == LLCameraPositionFront) {
+        AVCaptureConnection *videoConnection = [_movieFileOutput connectionWithMediaType:AVMediaTypeVideo];
+        if ([videoConnection isVideoMirroringSupported]) {
+            [videoConnection setVideoMirrored:YES];
```
Comment (omergul123) [inline LLSimpleCamera/LLSimpleCamera.m]: Why are you setting mirrored to **YES** ? shouldn't we check the **self.mirror*** property?

File: LLSimpleCamera/LLSimpleCamera.m [hunk 2]
```diff
@@ -527,6 +578,20 @@ - (void)setCameraPosition:(LLCameraPosition)cameraPosition
...
+        }
+
+        AVCaptureConnection *pictureConnection = [_stillImageOutput connectionWithMediaType:AVMediaTypeVideo];
+        if ([pictureConnection isVideoMirroringSupported]) {
+            [pictureConnection setVideoMirrored:YES];
```
Comment (omergul123) [inline LLSimpleCamera/LLSimpleCamera.m]: same here

**Comment by siong1987** [comment]
you are right.  we actually don't need those lines anymore.

**Comment by omergul123** [comment]
@siong1987 thanks! finally it's merged!

**Comment by siong1987** [comment]
@omergul123 :)

**Pull request merged**